kv/kvserver: TestStoreRangeSplitMergeStats failed

kv/kvserver.TestStoreRangeSplitMergeStats [failed](https://tanzanite.cluster.engflow.com/invocations/default/91fc29d1-24a0-47b0-bf81-0c244bc36eaf?testReportRun=3&testReportShard=36&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-24.2.0-rc @ [acb89a54d7923817d5fbdc11cadc91c60fe4345f](https://github.com/cockroachdb/cockroach/commits/acb89a54d7923817d5fbdc11cadc91c60fe4345f):


```
=== RUN   TestStoreRangeSplitMergeStats
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStoreRangeSplitMergeStats607211236
    test_log_scope.go:81: use -show-logs to present logs inline
    client_split_test.go:816: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/client_split_test.go:816
        	Error:      	Not equal: 
        	            	expected: enginepb.MVCCStats{ContainsEstimates:0, LastUpdateNanos:1724496147293235116, LockAge:0, GCBytesAge:139000, LiveBytes:18384, LiveCount:119, KeyBytes:19820, KeyCount:851, ValBytes:127677, ValCount:950, IntentBytes:0, IntentCount:0, LockBytes:0, LockCount:0, RangeKeyCount:98, RangeKeyBytes:13136, RangeValCount:1239, RangeValBytes:0, SysBytes:239, SysCount:5, AbortSpanBytes:0}
        	            	actual  : enginepb.MVCCStats{ContainsEstimates:0, LastUpdateNanos:1724496147293235116, LockAge:0, GCBytesAge:139000, LiveBytes:18384, LiveCount:119, KeyBytes:19820, KeyCount:851, ValBytes:127677, ValCount:950, IntentBytes:0, IntentCount:0, LockBytes:0, LockCount:0, RangeKeyCount:98, RangeKeyBytes:13136, RangeValCount:1239, RangeValBytes:0, SysBytes:233, SysCount:5, AbortSpanBytes:0}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -19,3 +19,3 @@
        	            	  RangeValBytes: (int64) 0,
        	            	- SysBytes: (int64) 239,
        	            	+ SysBytes: (int64) 233,
        	            	  SysCount: (int64) 5,
        	Test:       	TestStoreRangeSplitMergeStats
        	Messages:   	in-memory and on-disk stats diverge
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestStoreRangeSplitMergeStats607211236
--- FAIL: TestStoreRangeSplitMergeStats (4.58s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=36</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeSplitMergeStats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41668

--- a/pkg/kv/kvserver/client_split_test.go
+++ b/pkg/kv/kvserver/client_split_test.go
@@ -808,12 +808,19 @@ func TestStoreRangeSplitMergeStats(t *testing.T) {
 	start := s.Clock().Now()
 
 	// Get the range stats now that we have data.
+	// NB: need raft lock so that in-mem and disk stats can be expected to match.
+	// See: https://github.com/cockroachdb/cockroach/issues/129601#issuecomment-2309865742
+	repl.RaftLock()
+	replMS := repl.GetMVCCStats()
 	snap := store.TODOEngine().NewSnapshot()
 	defer snap.Close()
+	repl.RaftUnlock()
+
 	ms, err := stateloader.Make(repl.RangeID).LoadMVCCStats(ctx, snap)
 	require.NoError(t, err)
 	assertRecomputedStats(t, "before split", snap, repl.Desc(), ms, start.WallTime)
-	require.Equal(t, repl.GetMVCCStats(), ms, "in-memory and on-disk stats diverge")
+
+	require.Equal(t, replMS, ms, "in-memory and on-disk stats diverge")
 
 	// Split the range at approximate halfway point.
 	// Call AdminSplit on the replica directly so that we can pass a

